server/cache.go

```diff
@@ -663,15 +663,15 @@ func (c *clusterInfo) updateStoreStatus(id uint64) {
 
 // handleRegionHeartbeat updates the region information.
 func (c *clusterInfo) handleRegionHeartbeat(region *RegionInfo) error {
-	c.Lock()
-	defer c.Unlock()
-
 	region = region.clone()
+	c.RLock()
 	origin := c.regions.getRegion(region.GetId())
+	c.RUnlock()
```

Comment:
The cache and kv may not be consistent, for example, if a region's leader sends a heartbeat and then elects another leader and sends another heartbeat with a new version. But it seems not a problem, because they will be consistent finally?